crossbeam-utils/src/thread.rs

```diff
@@ -522,6 +522,35 @@ impl<'scope, T> ScopedJoinHandle<'scope, T> {
     }
 }
 
+#[cfg(unix)]
+use std::os::unix::thread::{JoinHandleExt, RawPthread};
+#[cfg(unix)]
+impl<'scope, T> JoinHandleExt for ScopedJoinHandle<'scope, T> {
+    fn as_pthread_t(&self) -> RawPthread {
+        self.handle.lock().unwrap().as_ref().unwrap().as_pthread_t()
```

Comment:
How about splitting this into two lines (as done for Windows)?

Reply:
I'm concerned with the safety here.  Is it okay to return a resource protected by a lock?  If so, why?

Reply:
I believe this is safe since the resource being returned (`RawPthread`) is an opaque handle to the underlying OS thread that is copied and returned to the caller. I don't see how this operation could affect the soundness of any other code.

Reply:
I've split the lines